lib/config/openssh.go

```diff
@@ -0,0 +1,214 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package config
+
+import (
+	"bytes"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+var sshConfigTemplate = template.Must(template.New("ssh-config").Parse(
+	`# Begin generated Teleport configuration for {{ .ProxyHost }} by {{ .AppName }}
+{{$dot := . }}
+{{- range $clusterName := .ClusterNames }}
+# Common flags for all {{ $clusterName }} hosts
+Host *.{{ $clusterName }} {{ $dot.ProxyHost }}
+    UserKnownHostsFile "{{ $dot.KnownHostsPath }}"
+    IdentityFile "{{ $dot.IdentityFilePath }}"
+    CertificateFile "{{ $dot.CertificateFilePath }}"{{- if $dot.NewerHostKeyAlgorithmsSupported }}
+    HostKeyAlgorithms rsa-sha2-256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com{{- else }}
```

Comment:
No, because of 3rd party libraries that I don't want to mention here. I changed the logic to work with SHA-2 and SHA-1 if the first one is not supported.